benchmark/result/metadata_class.py

```diff
@@ -0,0 +1,41 @@
+"""This module contains the metadata of a benchmark result."""
+
+from datetime import datetime
+
+import git
+from attrs import define, field
+from attrs.validators import instance_of
+
+from baybe import __version__ as baybe_version
+from baybe.serialization.mixin import SerialMixin
+
+
+@define(frozen=True)
+class ResultMetadata(SerialMixin):
+    """The metadata of a benchmark result."""
+
+    benchmark_name: str = field(validator=instance_of(str))
+    """The name of the benchmark."""
+
+    execution_time_sec: float = field(validator=instance_of(float))
+    """The execution time of the benchmark in seconds."""
+
+    gpu_used: bool = field(validator=instance_of(bool))
+    """Whether the benchmark used a GPU."""
```

Comment:
I suggest to skip this logic for now because the whole GPU part is not yet ready and who knows what other unforeseen blockers there will be that might require us to treat gpu usage differently in the benchmarks